methods.rb

```diff
@@ -0,0 +1,157 @@
+require_relative './app'
+require_relative './person'
+require_relative './student'
+require_relative './teacher'
+require_relative './classroom'
+require_relative './rental'
+require_relative './book'
+
+def list_books
+  print "List of all books:\n\n"
+  if @books.empty?
+    puts "Sorry, no books available at this time.\n"
+    sleep(2)
+    run
+  else
+    @books.each_with_index { |book, idx| print "#{idx + 1}) Title: #{book.title}, Author: #{book.author}\n" }
+    sleep(2)
+  end
+  run
+end
+
+def add_book
+  puts 'Please, enter title:'
+  title = gets.chomp
+  puts 'Please, enter author:'
+  author = gets.chomp
+  book = Book.new(title, author)
+  @books << book
+  puts "\n"
+  puts 'Congratulations, you created a book!!!'
+  puts "\n"
+  sleep(2)
+  run
+end
+
+def add_person
+  puts 'Please, enter age:'
+  age = gets.chomp.to_i
+  if age < 18
+    puts 'Age is below 18. Does this person has parent permission (y/n)?:'
+    permission = gets.chomp
+    case permission
+    when 'n'
+      puts 'Sorry, it is not possible to create this person.'
+      sleep(2)
+      run
+    when 'y'
+      enroll(age)
+    else
+      puts 'Please, select "y" or "n".'
+      sleep(2)
+      run
+    end
+  else
+    enroll(age)
+  end
+end
+
+def enroll(age)
+  puts 'Please enter 1 to create a teacher, 2 to create a student:'
+  opc = gets.chomp
+  case opc
+  when '1'
+    enroll_teacher(age)
+  when '2'
+    enroll_student(age)
+  else
+    puts 'Please select 1 or 2, no other options available.'
+    sleep(2)
+  end
+  run
+end
+
+def enroll_teacher(age)
+  puts 'Please enter name:'
+  name = gets.chomp
+  puts 'Please enter teacher specialization:'
+  specialization = gets.chomp
+  teacher = Teacher.new(age, specialization, name)
+  @people << teacher
```

Comment:
- [ ] Hi. As part of the SOLID Principle, this part should've been placed in your `add_person` method